packages/health-check/src/utils.js

```diff
@@ -39,9 +41,60 @@ function ensureValidJSON(object) {
   return JSON.parse(stringified);
 }
 
+/**
+ * Authenticate with given credentials and return auth cookie
+ * Creates new account if username does not exist
+ * Only authenticates when portal is set to authenticated users only mode
+ */
+function getAuthCookie() {
+  // cache auth promise so only one actual request will be made
+  if (getAuthCookie.cache) return getAuthCookie.cache;
+
+  // do not authenticate if it is not necessary
+  if (process.env.ACCOUNTS_LIMIT_ACCESS !== "authenticated") return {};
+
+  const email = process.env.ACCOUNTS_TEST_USER_EMAIL;
+  const password = process.env.ACCOUNTS_TEST_USER_PASSWORD;
+
+  if (!email) throw new Error("ACCOUNTS_TEST_USER_EMAIL cannot be empty");
+  if (!password) throw new Error("ACCOUNTS_TEST_USER_PASSWORD cannot be empty");
+
+  async function authenticate() {
+    try {
+      // authenticate with given test credentials
+      const response = await got.post(`${process.env.SKYNET_DASHBOARD_URL}/api/login`, {
+        json: { email, password },
+      });
+
+      // extract set-cookie from successful authentication request
+      const cookies = response.headers["set-cookie"];
+
+      // find the skynet-jwt cookie
+      const jwtcookie = cookies.find((cookie) => cookie.startsWith("skynet-jwt"));
```

Comment:
isn't the result of `.find` `null` or `undefined`, so won't you run into trouble calling `.match` on it later?

Reply:
well if we run into that then we're gonna throw an error and have to debug why we didn't get a cookie on successful authentication request :)

Reply:
lol, ok fine I'lll approve it but then you should
```
if (!jwtcookie) {
    throw new Error("'skynet-jwt' cookie not found');
}
```
}